ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/json/IsJsonString.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udf.json;
+
+import io.confluent.ksql.function.FunctionCategory;
+import io.confluent.ksql.function.KsqlFunctionException;
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+import io.confluent.ksql.function.udf.UdfParameter;
+import io.confluent.ksql.util.KsqlConstants;
+
+@UdfDescription(
+    name = "IS_JSON_STRING",
+    category = FunctionCategory.JSON,
+    description = "Given a string, returns true if it can be parsed as a valid JSON value, false"
+        + " otherwise.",
+    author = KsqlConstants.CONFLUENT_AUTHOR)
+public class IsJsonString {
+
+  @Udf
+  public Boolean check(@UdfParameter(description = "The input JSON string") final String input) {
+    if (input == null) {
+      return false;
+    }
+
+    try {
+      return !UdfJsonMapper.parseJson(input).isMissingNode();
+    } catch (KsqlFunctionException e) {
+      return false;
+    }
+  }
```

Comment:
Should we return true at the end?

Reply:
Can you please elaborate the reasoning behind your suggestion? As I see it, this function return `true` only if the string was parsed as JSON without exception and the underlying node is not `MissingNode` that represents broken JSON.

In any case, I think it is worth switching to a primitive type to signal that the return value is never `null` - https://github.com/confluentinc/ksql/pull/8600/commits/2f556c963b4f73e8242ea74ab4d75dce161b1d99#diff-48af5e77f4b3c1e1ee5cb1d402f247c3e70b4649c1de07e5171f8526dd0d908aR34.

Reply:
I am going to merge this PR as a bunch of other patches are blocked on it. We can continue the discussion here; I'll do a follow-up patch if necessary.

Reply:
Sorry my bad, miss-read the line of `.parseJson(input).isMissingNode();` please ignore :P